`matrix-synapse` build fails with the following:

```
===============================================================================
[FAIL]
Traceback (most recent call last):
  File "/build/source/tests/rest/client/test_login.py", line 1316, in test_login_aud_no_config
    self.assertEqual(
  File "/nix/store/pj3pzycszazvw73v66a2yq3k11hld846-python3.12-twisted-24.11.0/lib/python3.12/site-packages/twisted/trial/_synctest.py", line 444, in assertEqual
    super().assertEqual(first, second, msg)
  File "/nix/store/ic0fd07z8wg57idmw7q7iy7dc7md6x72-python3-3.12.9/lib/python3.12/unittest/case.py", line 885, in assertEqual
    assertion_func(first, second, msg=msg)
  File "/nix/store/ic0fd07z8wg57idmw7q7iy7dc7md6x72-python3-3.12.9/lib/python3.12/unittest/case.py", line 1251, in assertMultiLineEqual
    self.fail(self._formatMessage(msg, standardMsg))
twisted.trial.unittest.FailTest: "JWT validation failed: invalid_claim: Invalid claim 'aud'" != 'JWT validation failed: invalid_claim: Invalid claim "aud"'
- JWT validation failed: invalid_claim: Invalid claim 'aud'
?                                                     ^   ^
+ JWT validation failed: invalid_claim: Invalid claim "aud"
?                                                     ^   ^


tests.rest.client.test_login.JWTTestCase.test_login_aud_no_config
===============================================================================
[FAIL]
Traceback (most recent call last):
  File "/build/source/tests/rest/client/test_login.py", line 1297, in test_login_aud
    self.assertEqual(
  File "/nix/store/pj3pzycszazvw73v66a2yq3k11hld846-python3.12-twisted-24.11.0/lib/python3.12/site-packages/twisted/trial/_synctest.py", line 444, in assertEqual
    super().assertEqual(first, second, msg)
  File "/nix/store/ic0fd07z8wg57idmw7q7iy7dc7md6x72-python3-3.12.9/lib/python3.12/unittest/case.py", line 885, in assertEqual
    assertion_func(first, second, msg=msg)
  File "/nix/store/ic0fd07z8wg57idmw7q7iy7dc7md6x72-python3-3.12.9/lib/python3.12/unittest/case.py", line 1251, in assertMultiLineEqual
    self.fail(self._formatMessage(msg, standardMsg))
twisted.trial.unittest.FailTest: "JWT validation failed: invalid_claim: Invalid claim 'aud'" != 'JWT validation failed: invalid_claim: Invalid claim "aud"'
- JWT validation failed: invalid_claim: Invalid claim 'aud'
?                                                     ^   ^
+ JWT validation failed: invalid_claim: Invalid claim "aud"
?                                                     ^   ^


tests.rest.client.test_login.JWTTestCase.test_login_aud
===============================================================================
[FAIL]
Traceback (most recent call last):
  File "/build/source/tests/rest/client/test_login.py", line 1265, in test_login_iss
    self.assertEqual(
  File "/nix/store/pj3pzycszazvw73v66a2yq3k11hld846-python3.12-twisted-24.11.0/lib/python3.12/site-packages/twisted/trial/_synctest.py", line 444, in assertEqual
    super().assertEqual(first, second, msg)
  File "/nix/store/ic0fd07z8wg57idmw7q7iy7dc7md6x72-python3-3.12.9/lib/python3.12/unittest/case.py", line 885, in assertEqual
    assertion_func(first, second, msg=msg)
  File "/nix/store/ic0fd07z8wg57idmw7q7iy7dc7md6x72-python3-3.12.9/lib/python3.12/unittest/case.py", line 1251, in assertMultiLineEqual
    self.fail(self._formatMessage(msg, standardMsg))
twisted.trial.unittest.FailTest: "JWT validation failed: invalid_claim: Invalid claim 'iss'" != 'JWT validation failed: invalid_claim: Invalid claim "iss"'
- JWT validation failed: invalid_claim: Invalid claim 'iss'
?                                                     ^   ^
+ JWT validation failed: invalid_claim: Invalid claim "iss"
?                                                     ^   ^


tests.rest.client.test_login.JWTTestCase.test_login_iss
-------------------------------------------------------------------------------
Ran 4015 tests in 111.460s

FAILED (skips=166, failures=3, successes=3846)
```

I think it's a quotation difference in the error error messages returned by `authlib`, and the `matrix-synapse` test suite hardcoding one of them. This probably should be changed to not directly compare strings upstream, I assume other distros will run into this as well once bumping authlib.

--- a/nixos/tests/canaille.nix
+++ b/nixos/tests/canaille.nix
@@ -56,7 +56,7 @@ import ./make-test-python.nix (
         server.succeed("sudo -iu canaille -- canaille create user --user-name admin --password adminpass --emails admin@${domain}")
         json_str = server.succeed("sudo -iu canaille -- canaille get user")
         assert json.loads(json_str)[0]["user_name"] == "admin"
-        server.succeed("sudo -iu canaille -- canaille check")
+        server.succeed("sudo -iu canaille -- canaille config check")
       '';
   }
 )

--- a/pkgs/by-name/ca/canaille/package.nix
+++ b/pkgs/by-name/ca/canaille/package.nix
@@ -2,6 +2,7 @@
   lib,
   python3,
   fetchFromGitLab,
+  fetchpatch,
   openldap,
   nixosTests,
 }:
@@ -11,7 +12,7 @@ let
 in
 python.pkgs.buildPythonApplication rec {
   pname = "canaille";
-  version = "0.0.57";
+  version = "0.0.74";
   pyproject = true;
 
   disabled = python.pythonOlder "3.10";
@@ -20,8 +21,16 @@ python.pkgs.buildPythonApplication rec {
     owner = "yaal";
     repo = "canaille";
     rev = "refs/tags/${version}";
-    hash = "sha256-pesN7k5kGHi3dqTMaXWdCsNsnaJxXv/Ku1wVC9N9a3k=";
+    hash = "sha256-FL02ADM7rUU43XR71UWr4FLr/NeUau7zRwTMOSFm1T4=";
   };
+
+  patches = [
+    # https://gitlab.com/yaal/canaille/-/merge_requests/275
+    (fetchpatch {
+      url = "https://gitlab.com/yaal/canaille/-/commit/1c7fc8b1034a4423f7f46ad8adeced854910b702.patch";
+      hash = "sha256-fu7D010NG7yUChOve7HY3e7mm2c/UGpfcTAiTU8BnGg=";
+    })
+  ];
 
   build-system = with python.pkgs; [
     hatchling
@@ -32,10 +41,12 @@ python.pkgs.buildPythonApplication rec {
   dependencies =
     with python.pkgs;
     [
+      blinker
       flask
+      flask-caching
       flask-wtf
       pydantic-settings
-      requests
+      httpx
       wtforms
     ]
     ++ sentry-sdk.optional-dependencies.flask;
@@ -57,6 +68,7 @@ python.pkgs.buildPythonApplication rec {
       toml
       faker
       time-machine
+      pytest-scim2-server
     ]
     ++ optional-dependencies.front
     ++ optional-dependencies.oidc
@@ -79,27 +91,34 @@ python.pkgs.buildPythonApplication rec {
     export SCHEMA="${openldap}/etc/schema"
 
     # Just use their example config for testing
-    export CONFIG=canaille/config.sample.toml
+    export CONFIG=tests/app/fixtures/default-config.toml
   '';
 
   optional-dependencies = with python.pkgs; {
     front = [
       email-validator
       flask-babel
+      flask-talisman
       flask-themer
       pycountry
       pytz
-      toml
+      tomlkit
       zxcvbn-rs-py
     ];
-    oidc = [ authlib ];
+    oidc = [
+      authlib
+      joserfc
+    ];
     scim = [
+      httpx
       scim2-models
       authlib
+      scim2-client
     ];
     ldap = [ python-ldap ];
     sentry = [ sentry-sdk ];
     postgresql = [
+      flask-alembic
       passlib
       sqlalchemy
       sqlalchemy-json
@@ -111,6 +130,7 @@ python.pkgs.buildPythonApplication rec {
       qrcode
     ];
     sms = [ smpplib ];
+    server = [ hypercorn ];
   };
 
   passthru = {
@@ -119,16 +139,6 @@ python.pkgs.buildPythonApplication rec {
       inherit (nixosTests) canaille;
     };
   };
-
-  disabledTests = [
-    # cause by authlib being too up-to-date for this version of canaille
-    # see: https://github.com/NixOS/nixpkgs/issues/389861#issuecomment-2726361949
-    # FIX: update and see if this is fixed
-    "test_invalid_client[ldap_backend]"
-    "test_invalid_client[memory_backend]"
-    "test_invalid_client[sql_backend]"
-    "test_password_reset[sql_backend]"
-  ];
 
   meta = with lib; {
     description = "Lightweight Identity and Authorization Management";

--- a/pkgs/development/python-modules/authlib/default.nix
+++ b/pkgs/development/python-modules/authlib/default.nix
@@ -20,7 +20,7 @@
 
 buildPythonPackage rec {
   pname = "authlib";
-  version = "1.5.1";
+  version = "1.5.2";
   pyproject = true;
 
   disabled = pythonOlder "3.8";
@@ -29,7 +29,7 @@ buildPythonPackage rec {
     owner = "lepture";
     repo = "authlib";
     tag = "v${version}";
-    hash = "sha256-VMihaWqR4FbnTJ50fVf5e5B9GfVwRguq5UAC+D4bpxs=";
+    hash = "sha256-ra1RKprUAqhax0z1osl0lFgFENQZuSW/5FxSmsCdKNY=";
   };
 
   build-system = [ setuptools ];

--- a/pkgs/development/python-modules/flask-alembic/default.nix
+++ b/pkgs/development/python-modules/flask-alembic/default.nix
@@ -1,0 +1,52 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  pythonAtLeast,
+  flit-core,
+  alembic,
+  flask,
+  sqlalchemy,
+  pytestCheckHook,
+  flask-sqlalchemy,
+  flask-sqlalchemy-lite,
+}:
+
+buildPythonPackage rec {
+  pname = "flask-alembic";
+  version = "3.1.1";
+
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "pallets-eco";
+    repo = "flask-alembic";
+    tag = version;
+    hash = "sha256-iHJr9l3w1WwZXDl573IV7+A7RDcawGL20sxxhAQQ628=";
+  };
+
+  build-system = [ flit-core ];
+
+  dependencies = [
+    alembic
+    flask
+    sqlalchemy
+  ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    flask-sqlalchemy
+    flask-sqlalchemy-lite
+  ];
+
+  pythonImportChecks = [ "flask_alembic" ];
+
+  meta = with lib; {
+    # https://github.com/pallets-eco/flask-alembic/issues/47
+    broken = pythonAtLeast "3.13";
+    homepage = "https://github.com/pallets-eco/flask-alembic";
+    changelog = "https://github.com/pallets-eco/flask-alembic/blob/${src.tag}/CHANGES.md";
+    license = licenses.mit;
+    maintainers = with maintainers; [ erictapen ];
+  };
+}

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -4981,6 +4981,8 @@ self: super: with self; {
 
   flask-admin = callPackage ../development/python-modules/flask-admin { };
 
+  flask-alembic = callPackage ../development/python-modules/flask-alembic { };
+
   flask-allowed-hosts = callPackage ../development/python-modules/flask-allowed-hosts { };
 
   flask-api = callPackage ../development/python-modules/flask-api { };

